src/sigal/plugins/lrp.py

```diff
@@ -0,0 +1,47 @@
+"""Plugin which add raw support.
+
+Plugin make temp directory which is use for generate jpeg from raw file (jpeg is camera-generated thumbnail).
+For this dcraw program is used.
+
+"""
+
+import logging
+
+from sigal import signals
+
+import os
+import sys
+import re
+
+from ..gallery import Image
+from ..settings import Status, get_thumb
+import copy
+
+logger = logging.getLogger(__name__)
+
+tempDir="/tmp/rawTempSigal"
+
+def pouTest(gallery):
+
+	if gallery.src_ext == '.orf':
+		logger.info('Processing RAW: '+gallery.dst_filename)
+		newFilename=re.sub('ORF', 'jpeg', gallery.dst_filename)
+		tmpname = tempDir + '/'+newFilename
+
+		os.system('dcraw -c -e "' + gallery.src_path + '" > "' + tmpname + '"')
```

Comment:
If you use subprocess you can check return codes, this is just a run it and hope it succeeds.

Reply:
It is also not a safe way to provide arguments to a program, provide each part as an element of a list.

>  Providing a sequence of arguments is generally preferred, as it allows the module to take care of any required escaping and quoting of arguments (e.g. to permit spaces in file names).